src/components/SchedulePanel/components/OfficeHoursDialog/index.js

```diff
@@ -0,0 +1,526 @@
+              locationInput: eventDescLoc[1],
+              descriptionValid: true,
+              locationValid: true,
+              startHourValid: true,
+              endHourValid: true,
+              formValid: true,
+              checkedValid: true,
+            });
+          })
+          .catch(error => {
+            console.log(error);
+          });
+      });
+      // Case : Add(selecslot)
+    } else {
+      this.handleReset();
+      this.setState({
+        checkedDayofWeek: {
+          checkedSu: nextProps.resourceId === 1 ? true : false,
+          checkedMo: nextProps.resourceId === 2 ? true : false,
+          checkedTu: nextProps.resourceId === 3 ? true : false,
+          checkedWe: nextProps.resourceId === 4 ? true : false,
+          checkedTh: nextProps.resourceId === 5 ? true : false,
+          checkedFr: nextProps.resourceId === 6 ? true : false,
+          checkedSa: nextProps.resourceId === 7 ? true : false,
+        },
+        startHourInput:nextProps.startTime,
+        endHourInput:nextProps.endTime,
+        checkedValid: true,
+        startHourValid: true,
+        endHourValid: true,
+      });
+    }
+  }
+
+  validateField(fieldName, value) {
+    let fieldValidationErrors = this.state.formErrors;
+    let descriptionValid = this.state.descriptionValid;
+    let locationValid = this.state.locationValid;
+    let startHourValid = this.state.startHourValid;
+    let endHourValid = this.state.endHourValid;
+
+    // Require that content havint appropriate size and format
+    switch (fieldName) {
+      case 'descriptionInput':
+        descriptionValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.descriptionInput = descriptionValid ? '' : 'Description is not Valid';
+        break;
+      case 'locationInput':
+        locationValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.locationInput = locationValid ? '' : 'Location is not Valid';
+        break;
+      case 'startHourInput':
+        startHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.endHourInput, 10) > parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 10;
+        fieldValidationErrors.startHourInput = startHourValid ? '' : 'Start Time is not Valid';
+        break;
+      case 'endHourInput':
+        endHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.startHourInput, 10) < parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 22;
+        fieldValidationErrors.endHourInput = endHourValid ? '' : 'End Time is not Valid';
+        break;
+      default:
+        break;
+    }
+    this.setState(
+      {
+        formErrors: fieldValidationErrors,
+        descriptionValid: descriptionValid,
+        locationValid: locationValid,
+        startHourValid: startHourValid,
+        endHourValid: endHourValid,
+      },
+      () => {
```

Comment:
Again, is there a reason to run validateForm within the setState? Why not just make it the next function?